ingestion/src/metadata/profiler/metrics/core.py

```diff
@@ -153,6 +159,13 @@ def metric_type(self):
         """
         return self.col.type.python_type if self.col else None
 
+    def nosql_fn(self, client: NoSQLAdaptor) -> Callable[[Table], Optional[T]]:
+        """
+        Return the function to be used for NoSQL clients to calculate the metric.
+        By default, returns a "do nothing" function that returns None.
+        """
+        return lambda table: None
```

Comment:
why do we need to return a function and not just `None`?

Reply:
The method is returning a `Callable`:

https://github.com/open-metadata/OpenMetadata/pull/15189/files/74c2abd3de021f11d8b1630399605288260a007d..1733b9af92e92f317bd518855337da94f6699b76#diff-e439fc0df5e00e522353c3ee002acdf38e1320efcd148877a5ed01accee72279R162

 If it returns a bare `None` it will need to handle a branch to to avoid `TypeError: 'NoneType' object is not callable`.